ribbon/src/main/java/feign/ribbon/RibbonClient.java

```diff
@@ -69,13 +69,23 @@ public Response execute(Request request, Request.Options options) throws IOExcep
       return lbClient(clientName).executeWithLoadBalancer(ribbonRequest,
           new FeignOptionsClientConfig(options)).toResponse();
     } catch (ClientException e) {
-      if (e.getCause() instanceof IOException) {
-        throw IOException.class.cast(e.getCause());
-      }
+      checkForIOException(e);
       throw new RuntimeException(e);
     }
   }
 
+  static void checkForIOException(Throwable throwable) throws IOException {
+
+    final List<Throwable> list = new ArrayList<Throwable>(); // for detecting cycles
```

Comment:
it is handled in apache commons lang. the guava Throwables methods (at least in version 14 included by the project) are marked as Beta.